contracts/drafts/voting/Voting.sol

```diff
@@ -0,0 +1,167 @@
+    using FixidityLib for int256;
+    using SafeCast for int256;
+    using SafeCast for uint256;
+
+    event VotingCreated();
+    event QuorumSet();
+    event VoteCasted(address voter, uint256 votes);
+    event VoteCancelled(address voter, uint256 votes);
+
+    address public votingToken;
+    address public proposalContract;
+    bytes public proposalData;
+
+    address[] public voters;
+    mapping(address => uint256) public votes;
+
+    uint256 public quorum;
+
+    /**
+     * @dev Initialize the issuance with the token to issue and the token to
+     * accept as payment.
+     */
+    constructor(
+        address _votingToken,
+        address _proposalContract,
+        bytes memory _proposalData
+    ) public Ownable() StateMachine() {
+        votingToken = _votingToken;
+        proposalContract = _proposalContract;
+        proposalData = _proposalData;
+        _createState("OPEN");
+        _createState("PASSED");
+        _createState("FAILED");
+        _createTransition("SETUP", "OPEN");
+        _createTransition("OPEN", "PASSED");
+        _createTransition("OPEN", "FAILED");
+        emit VotingCreated();
+    }
+
+    /**
+     * @dev Function to enact the proposal of this voting
+     */
+    function enact() external payable onlyOwner {
+        require(
+            currentState == "PASSED",
+            "Cannot enact proposal now."
+        );
+        // solium-disable-next-line security/no-call-value
+        (bool success, ) = proposalContract
+            .call.value(msg.value)(proposalData);
+        require(success, "Could not enact proposal.");
+    }
+
+    /**
+     * @dev Use this function to cast votes. Must have approved this contract
+     * (from the frontend) to spend _votes of votingToken tokens.
+     * @param _votes The amount of votingToken tokens that will be casted.
+     */
+    function cast(uint256 _votes) external {
+        require(
+            currentState == "OPEN",
+            "Not open for voting."
+        );
+        IERC20(votingToken).transferFrom(msg.sender, address(this), _votes);
+        if (votes[msg.sender] == 0){
+            voters.push(msg.sender);
+        }
+        votes[msg.sender] = votes[msg.sender].add(_votes);
+        emit VoteCasted(msg.sender, _votes);
+    }
+
+    /**
+     * @dev Use this function to retrieve your votingToken votes in case you changed your mind.
+     */
+    function cancel() external {
+        require(
+            currentState != "SETUP",
+            "Cannot retrieve now."
+        );
+        require(
+            votes[msg.sender] > 0,
+            "No votes casted."
+        );
+        uint256 count = votes[msg.sender];
+        votes[msg.sender] = 0;
+        IERC20(votingToken).transfer(msg.sender, count);
+        emit VoteCancelled(msg.sender, count);
+    }
+
+    /**
+     * @dev Function to open the voting
+     */
+    function open() public onlyOwner {
+        require(
+            quorum > 0,
```

Comment:
Please rename "[quorum](https://en.wikipedia.org/wiki/Quorum)" to "[threshold](https://en.wikipedia.org/wiki/Electoral_threshold)" throughout.